File: app/src/main/java/com/remotearthsolutions/expensetracker/services/GoogleService.java
package com.remotearthsolutions.expensetracker.services;

import android.content.Intent;
import com.google.android.gms.auth.api.signin.GoogleSignInClient;

public interface GoogleService {
    void initializeGoogleSigninClient();
    void startGoogleLogin(Intent data,Callback callback);
    GoogleSignInClient getGoogleSignInClient();

    interface Callback extends SocialLoginCallback{
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/services/GoogleService.kt
package com.remotearthsolutions.expensetracker.services

import android.content.Intent
import com.google.android.gms.auth.api.signin.GoogleSignInClient

interface GoogleService {
    fun initializeGoogleSigninClient()
    fun startGoogleLogin(
        data: Intent?,
        callback: Callback?
    )

    val googleSignInClient: GoogleSignInClient?

    interface Callback : SocialLoginCallback
}